workers/generic-worker/mounts_test.go

```diff
@@ -421,6 +421,63 @@ func TestFileMountNoSHA256(t *testing.T) {
 	)
 }
 
+func TestFileMountWithCompression(t *testing.T) {
+	setup(t)
+	taskID := CreateArtifactFromFile(t, "compressed-file-mount.txt.gz", "public/build/compressed-file-mount.txt.gz")
+
+	// whether permission is granted to task user depends if running under windows or not
+	// and is independent of whether running as current user or not
+	granting, _ := grantingDenying(t, "file", t.Name())
+
+	// No cache on first pass
+	pass1 := append([]string{
+		`Downloading task ` + taskID + ` artifact public/build/compressed-file-mount.txt.gz to .*`,
+		`Downloaded 89 bytes with SHA256 a37856e8cd10250f76dc076bb03d380b16a870dec31f3461223f753124a4b28a from task ` + taskID + ` artifact public/build/compressed-file-mount.txt.gz to .*`,
+		`Content from task ` + taskID + ` artifact public/build/compressed-file-mount.txt.gz .* matches required SHA256 a37856e8cd10250f76dc076bb03d380b16a870dec31f3461223f753124a4b28a`,
+		`Creating directory .* with permissions 0700`,
+		`Decompressing gz file .* to .*` + t.Name(),
+	},
+		granting...,
+	)
+
+	// On second pass, cache already exists
+	pass2 := append([]string{
+		`Found existing download for artifact:` + taskID + `:public/build/compressed-file-mount.txt.gz .* with correct SHA256 a37856e8cd10250f76dc076bb03d380b16a870dec31f3461223f753124a4b28a`,
+		`Creating directory .* with permissions 0700`,
+		`Decompressing gz file .* to .*` + t.Name(),
+	},
+		granting...,
+	)
+
+	LogTest(
+		&MountsLoggingTestCase{
+			Test: t,
+			Mounts: []MountEntry{
+				&FileMount{
+					File: t.Name(),
+					Content: json.RawMessage(`{
+						"taskId":   "` + taskID + `",
+						"artifact": "public/build/compressed-file-mount.txt.gz",
+						"sha256":	"a37856e8cd10250f76dc076bb03d380b16a870dec31f3461223f753124a4b28a"
+					}`),
+					Format: "gz",
+				},
+			},
+			Dependencies: []string{
+				taskID,
+			},
+			TaskRunResolutionState: "completed",
+			TaskRunReasonResolved:  "completed",
+			PerTaskRunLogExcerpts: [][]string{
+				// Required text from first task with no cached value
+				pass1,
+				// Required text from second task when download is already cached
+				pass2,
+			},
```

Comment:
Would be nice to set `Payload` member here too, to cat the mounted file, to check it was compressed.